include/seqan3/alignment/pairwise/alignment_algorithm.hpp

```diff
@@ -665,22 +676,18 @@ class alignment_algorithm :
      *
      * If the alignment is run in debug mode (see seqan3::align_cfg::debug) the debug score and optionally trace matrix
      * are stored in the alignment result as well.
+     *
+     * Finally, the callback is invoked with each computed alignment result iteratively.
      */
-    template <typename indexed_sequence_pair_range_t>
+    template <typename indexed_sequence_pair_range_t, typename callback_t>
     //!\cond
         requires traits_t::is_vectorised
     //!\endcond
-    constexpr auto make_alignment_result(indexed_sequence_pair_range_t && index_sequence_pairs)
+    constexpr auto make_alignment_result(indexed_sequence_pair_range_t && index_sequence_pairs,
+                                         callback_t & callback)
     {
-        using indexed_sequence_pair_t = std::ranges::range_value_t<indexed_sequence_pair_range_t>;
-        using sequence_pair_t = std::tuple_element_t<0, indexed_sequence_pair_t>;
-        using sequence1_t = std::tuple_element_t<0, sequence_pair_t>;
-        using sequence2_t = std::tuple_element_t<1, sequence_pair_t>;
-
-        using result_value_t = typename align_result_selector<sequence1_t, sequence2_t, config_t>::type;
-
-        std::vector<alignment_result<result_value_t>> results{};
-        results.reserve(std::ranges::distance(index_sequence_pairs));
+        using alignment_result_t = typename function_traits<callback_t>::template argument_type_at<0>;
```

Comment:
it is a little confusing that the `alignment_result_t` is not equal to the `result_value_t` as they sound like synonyms, but of course, this does not relate to the PR.

Reply:
but it will be anyway resolve once we use the builder pattern for the alignment as well